src/modules/esl-utils/dom/rect.ts

```diff
@@ -28,6 +28,21 @@ export class Rect {
     return rect1.x === rect2.x && rect1.y === rect2.y && rect1.width === rect2.width && rect1.height === rect2.height;
   }
 
+  /**
+   * The static method returns intersection Rect of two Rect instances
+   * @param rect1 - first instance of Rect
+   * @param rect2 - second instance of Rect
+   */
+  public static intersect(rect1: Rect, rect2: Rect): Rect {
+    const top = Math.max(rect1.top, rect2.top);
+    const left = Math.max(rect1.left, rect2.left);
```

Comment:
Probably should use `y` instead of `top` and `x` instead of `left` because they are aliases of `x` and `y` properties.

Reply:
This was discussed with @ala-n and we decided to keep top/left for consistency with bottom/right